src/core_plugins/vega/public/vega_view/vega_base_view.js

```diff
@@ -153,21 +153,23 @@ export class VegaBaseView {
 
   destroy() {
     // properly handle multiple destroy() calls by converting this._destroyHandlers
-    // from an array into a promise, while handlers are being disposed
+    // into the _ongoingDestroy promise, while handlers are being disposed
     if (this._destroyHandlers) {
-      if (this._destroyHandlers.then) {
-        return this._destroyHandlers;
-      } else {
-        this.setDebugValues();
-        this._destroyHandlers = Promise.all(this._destroyHandlers.map(v => v()));
-        return this._destroyHandlers.then(() => this._destroyHandlers = null);
-      }
+      // If no destroy is yet running, execute all handlers and wait for all of them to resolve.
+      // Once done, keep the resolved promise, and get rid of any values returned from handlers.
+      this._ongoingDestroy = Promise.all(this._destroyHandlers.map(v => v())).then(() => 0);
```

Comment:
What is the `.then(() => 0)` intended to do? It will just reset the resolve value to `0`. Is that really needed?

Reply:
@timroes I think I was over-thinking it :)  Basically if the handler returns an object, all these objects are stored in an array inside the promise that is part of the View. Setting it to 0 expedites garbage collection, but in reality i don't think it matters. I will remove it in the followup patch.